src/core/Akka.Streams.Tests/IO/FileSinkSpec.cs

```diff
@@ -209,6 +210,23 @@ public void SynchronousFileSink_should_allow_overriding_the_dispatcher_using_Att
             }, _materializer);
         }
 
+        // Needed help converting this test case
+        [Fact]
+        public void SynchronousFileSink_should_write_single_line_to_a_file_from_lazy_sink()
+        {
+            this.AssertAllStagesStopped(() => {
+                TargetFile(f => {
+                    var completion = Source.From(_testByteStrings);
+                    completion.RunWith(Sink.LazySink<ByteString, Task<IOResult>>(
+                            b => Task.FromResult(FileIO.ToFile(f)),
+                            () => Task.FromResult(new IOResult(0, new Result<NotUsed>(NotUsed.Instance)))
+                        ), _materializer);
+                    completion.MapMaterializedValue();
```

Comment:
this needs to be `completion.MapMaterializedValue(list => list.SelectMany(x => x));` not sure about the actual type of `list` but `flatMap` means `SelectMany`